pandas/plotting/_matplotlib/tools.py

```diff
@@ -288,6 +288,36 @@ def _remove_labels_from_axis(axis):
     axis.get_label().set_visible(False)
 
 
+def _has_externally_shared_axis(ax1, compare_axis):
+    """If two axes sharing an axis also have the same position, they can be referred to
+    as internally sharing the common axis (a.k.a twinning). _handle_shared_axes() is
+    only interested in axes externally sharing an axis, regardless of whether either of
+    the axes is also internally sharing with a third axes.
+
+    """
+
+    if compare_axis == "x":
+        axes = ax1.get_shared_x_axes()
+    elif compare_axis == "y":
+        axes = ax1.get_shared_y_axes()
+    else:
+        raise ValueError(
+            "_has_externally_shared_axis() needs 'x' or 'y' as a second parameter"
+        )
+
+    axes = axes.get_siblings(ax1)
+
+    # Remove colocated axes
+    ax1_points = ax1.get_position().get_points()
+    axes = [
+        ax2
+        for ax2 in axes
+        if ax1 == ax2 or not np.array_equal(ax1_points, ax2.get_position().get_points())
```

Comment:
Here, is the check that ax2 and ax1 are the same object? If so, I think ax2 is ax1 would be clearer